chips/rp2040/src/chip.rs

```diff
@@ -0,0 +1,196 @@
+//! Chip trait setup.
+
+use core::fmt::Write;
+use kernel::common::deferred_call;
+use kernel::Chip;
+use kernel::InterruptService;
+
+use crate::adc;
+use crate::clocks::Clocks;
+use crate::deferred_call_tasks::DeferredCallTask;
+use crate::gpio::{RPPins, SIO};
+use crate::interrupts;
+use crate::resets::Resets;
+use crate::timer::RPTimer;
+use crate::uart::Uart;
+use crate::watchdog::Watchdog;
+use crate::xosc::Xosc;
+use cortexm0p::interrupt_mask;
+
+#[repr(u8)]
+pub enum Processor {
+    Processor0 = 0,
+    Processor1 = 1,
+}
+
+pub struct Rp2040<'a, I: InterruptService<DeferredCallTask> + 'a> {
```

Comment:
Given that this itself chip doesn't actually need any deferred calls, you can throw the `deferred_call_tasks` module out and use a unit `()` for this type parameter as a placeholder instead, like done in `e310x`: https://github.com/tock/tock/blob/1b3c6a14cf22d3ec2959a22c69b69e336aac7903/chips/e310x/src/chip.rs#L18

Reply:
done